### DISCUSSION TITLE
Python drone connection Tmotor F7(Betaflight STM32F7x2)

### DISCUSSION THREAD

**Discussion by YY-71** [iNavFlight/inav-configurator#1999] [Q&A]
Hello everyone, 

I am doing a project to connect drone with microcontroller Betaflight STM32F7x2 with python code. After some trials, I succeed detect the serial port with STM but it failed to connect like normal. My target is simple, to connect the STM, get battery value and simple motor activation.

I have tried with dronekit library, but the error showed as line below
_class Parameters(collections.MutableMapping, HasObservers):
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'collections' has no attribute 'MutableMapping'_
-------------ori code----------
from dronekit import connect
vehicle = connect('COM6', wait_ready=True)
print("Mode: %s" % vehicle.mode.name)
--------------------------------
Besides, I have tried also with pymavlink, the error is different:
_Serial port COM6 opened successfully.
Error establishing MAVLink connection: invalid literal for int() with base 10: 'COM6'_
------------ori code-----------
import serial
from pymavlink import mavutil
def mavlink_connection(port, baudrate):
    try:
        connection = serial.Serial(port, baudrate)
        # Create MAVLink connection
        print(f"Serial port {port} opened successfully.")
        mavlink_connection = mavutil.mavlink_connection('serial:' + port, baud=baudrate)
        print("MAVLink connection established.")
        return mavlink_connection
    except Exception as e:
        print(f"Error establishing MAVLink connection: {e}")
        return None

if __name__ == "__main__":
    # Serial port settings
    port = 'COM6'
    baudrate = 115200
    # Establish MAVLink connection
    mavlink_conn = mavlink_connection(port, baudrate)
    if mavlink_conn:
        # Request heartbeat
        mavlink_conn.mav.heartbeat_send(mavutil.mavlink.MAV_TYPE_GCS, mavutil.mavlink.MAV_AUTOPILOT_INVALID, 0, 0, 0)
------------------------------------

Please comments if you any idea to solve it. Firstly for the connection. Thank you!

**Comment by sensei-hacker** [comment]
That's probably an old pipenv with new Python.
In any event, it's a Python issue.

https://github.com/pypa/pipenv/issues/5133
